src/main/scala/tech/cryptonomic/conseil/config/Security.scala

```diff
@@ -1,8 +1,46 @@
 package tech.cryptonomic.conseil.config
 
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport
 import pureconfig.generic.auto._
 
-object Security {
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object Security extends ErrorAccumulatingCirceSupport with LazyLogging {
+
+  /** Keys cache */
+  private val nautilusCloudKeys: Ref[IO, Set[String]] = Ref.of[IO, Set[String]](Set.empty).unsafeRunSync()
+
+  /** Updates API keys from Nautilus-Cloud endpoint */
+  def updateKeys(
+      ncc: NautilusCloudConfiguration
+  )(implicit executionContext: ExecutionContext, system: ActorSystem, mat: Materializer): Unit = {
+    val update = for {
+      apiKeys <- Http()
+        .singleRequest(
+          HttpRequest(uri = makeUri(ncc))
+            .withHeaders(RawHeader("X-Api-Key", ncc.key))
+        )
+        .flatMap(Unmarshal(_).to[Set[String]])
+      _ <- nautilusCloudKeys.update(_ => apiKeys).unsafeToFuture()
+    } yield logger.info(s"Managed to update api keys with $apiKeys")
+    update onComplete {
```

Comment:
I usually call `.andThen` to log failures, as it accepts a PartialFunction, so I can avoid writing an empty success case.

In this situation tho, I'd only log success if the outcome is thus, therefore:

```scala
val update = for {
  ...
} yield ()
update onComplete {
  case Success(_) => logger.info("Managed to update api keys with {}", apiKeys)
  case Failure(exception) => logger.error("Error during API keys update", exception)
}
```

Reply:
On a different note, my personal stance when writing log message, is that we should stop using messages that sounds like they're spit out from the first Terminator version or some '70s TV show.
Even Hal in Space Odissey conveys information like a human... I try to be follow its lead for a better user experience. 😁

Reply:
Wait a minute, both the terminator and HAL are homicidal. Now if we are using that as critical criteria, may i humbly put forth that we use Bender from Futurama as an exemplar. :)

Reply:
ok, changed. 

Regarding log messages. I like terminator way of logging. It is concise and brings only useful information. Did you have in mind something like this?: `Hello dear user. We've encountered some problems during API keys update request. But do not worry, here you can find what went wrong:` 😄 (I've watched Space Odyssey long time ago)

Reply:
@bhaskarkishore Bender? `Here you go, your request failed. I'll make my own request, with blackjack and hookers. In fact, forget the request` 🤣

Reply:
I was aiming more at something like the Downton Abbey's butler...

"Sir, I beg your pardon. It's quite embarrassing yet the updates you requested were temporarily unavailable, may I take care to provide you with a suitable alternative right away or has milord changed his mind?"